File: src/main/java/csp/backfront/model/entity/Address.java
package csp.backfront.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Past;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@AllArgsConstructor
@Builder
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Past
    private LocalDateTime createdDate;

    @NotBlank
    private String country;

    @NotBlank
    private String city;

    @NotBlank
    private String street;

    @NotBlank
    private String houseNumber;

    @NotBlank
    private String apartment;

    @ManyToOne
    private User user;

    public Address() {
    }
}

File: src/main/front/contacts/src/app/model/address.ts
export class Address {

  constructor(country: string, city: string, street: string, houseNumber: string, apartment: string) {
    this._country = country;
    this._city = city;
    this._street = street;
    this._houseNumber = houseNumber;
    this._apartment = apartment;
  }

  private _country: string;

  get country(): string {
    return this._country;
  }

  set country(value: string) {
    this._country = value;
  }

  private _city: string;

  get city(): string {
    return this._city;
  }

  set city(value: string) {
    this._city = value;
  }

  private _street: string;

  get street(): string {
    return this._street;
  }

  set street(value: string) {
    this._street = value;
  }

  private _houseNumber: string;

  get houseNumber(): string {
    return this._houseNumber;
  }

  set houseNumber(value: string) {
    this._houseNumber = value;
  }

  private _apartment: string;

  get apartment(): string {
    return this._apartment;
  }

  set apartment(value: string) {
    this._apartment = value;
  }
}
